shell/platform/embedder/embedder.h

```diff
@@ -541,6 +542,77 @@ typedef struct {
   FlutterMetalTextureFrameCallback external_texture_frame_callback;
 } FlutterMetalRendererConfig;
 
+/// Alias for VkInstance.
+typedef const uint64_t* FlutterVulkanInstanceHandle;
+
+/// Alias for VkPhysicalDevice.
+typedef const uint64_t* FlutterVulkanPhysicalDeviceHandle;
+
+/// Alias for VkDevice.
+typedef const uint64_t* FlutterVulkanDeviceHandle;
+
+/// Alias for VkImage.
+typedef const uint64_t* FlutterVulkanImageHandle;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterVulkanImage).
+  size_t struct_size;
+  /// Handle to the VkImage that is owned by the embedder. The engine will
+  /// bind this image for writing the frame.
+  FlutterVulkanImageHandle image;
+  /// A baton that is not interpreted by the engine in any way. It will be given
+  /// back to the embedder in the destruction callback below. Embedder resources
+  /// may be associated with this baton.
+  void* user_data;
+  /// The callback invoked by the engine when it no longer needs this backing
+  /// store.
+  VoidCallback destruction_callback;
+} FlutterVulkanImage;
+
+/// Callback to fetch a Vulkan function pointer for a given instance. Normally,
+/// this should return the results of vkGetInstanceProcAddr.
+typedef void* (*FlutterVulkanInstanceProcAddressCallback)(
+    void* /* user data */,
+    FlutterVulkanInstanceHandle /* instance */,
+    const char* /* name */);
+
+/// Callback for when a VkImage is requested.
+typedef FlutterVulkanImage (*FlutterVulkanImageCallback)(
+    void* /* user data */,
+    const FlutterFrameInfo* /* frame info */);
+
+/// Callback for when a VkImage has been written to and is ready for use by the
+/// embedder.
+typedef bool (*FlutterVulkanPresentCallback)(
+    void* /* user data */,
+    const FlutterVulkanImage* /* image */);
+typedef uint64_t* (*Uint64Callback)(void* /* user data */);
+
+typedef uint64_t* (*FlutterVulkanPresentImageCallback)(void* /* user data */,
+                                                       uint64_t* /* image */);
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterVulkanRendererConfig).
+  size_t struct_size;
+  // VkInstance handle.
+  FlutterVulkanInstanceHandle instance;
+  // VkPhysicalDevice handle.
+  FlutterVulkanPhysicalDeviceHandle physical_device;
+  // VkDevice handle.
+  FlutterVulkanDeviceHandle device;
+  ProcResolver get_proc_address_callback;
```

Comment:
Ah, this signature is used in the GL renderer. I think this was supposed to be instance-less vkGetInstanceProcAddr in the doc, but on second thought I don't see a reason why this would be useful as a separate callback. Removed.